lib/experimental/full-page-client-side-navigation.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Registers full page client-side navigation option using the Interactivity API and adds the necessary directives.
+ */
+
+// Set the navigation mode to full page client-side navigation.
+wp_interactivity_config( 'core/router', array( 'navigationMode' => 'fullPage' ) );
+
+// Enqueue the interactivity router script.
```

Comment:
I don't think that `init` is the correct hook because this script shouldn't get enqueued in WP Admin. 

Nitpick: for function signatures, PHPDoc format should be used:
```suggestion
/**
 * Enqueues the interactivity router script.
 */
```

Reply:
[`wp_enqueue_hooks`](https://developer.wordpress.org/reference/hooks/wp_enqueue_scripts/) seems like a proper hook to use here.

Reply:
Done in this commit: [link](https://github.com/WordPress/gutenberg/pull/59707/commits/181614e418fa958f90077d7a3e9a409326126299).